client/solve.go

```diff
@@ -125,12 +126,23 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
 		return nil, err
 	}
 
-	var ex ExportEntry
-	if len(opt.Exports) > 1 {
-		return nil, errors.New("currently only single Exports can be specified")
+	type exporter struct {
+		ExportEntry
+		id string
 	}
-	if len(opt.Exports) == 1 {
-		ex = opt.Exports[0]
+
+	var exporters []exporter
+	ids := make(map[string]int)
+	for _, exp := range opt.Exports {
+		if id, ok := ids[exp.Type]; !ok {
+			ids[exp.Type] = 1
+		} else {
+			ids[exp.Type] = id + 1
+		}
+		exporters = append(exporters, exporter{
+			ExportEntry: exp,
+			id:          fmt.Sprint(exp.Type, ids[exp.Type]),
```

Comment:
This looks deterministic enough but it would be useful if wer set an arbitrary id:

```
buildctl build ... \
  --output id=foo,type=image,name=docker.io/username/image,push=true \
  --output id=bar,type=local,dest=path/to/output-dir \
```

Then user could easily parse the result metadata output.

Reply:
@crazy-max see above, I've removed the concept of an explicit ID - WDYT?